src/components/navigation/document-tab-panel.tsx

```diff
@@ -26,15 +26,32 @@ interface IState {
 
 const kNavItemScale = 0.11;
 
+export interface ISubTabSpec {
+  label: string;
+  sections: NavTabSectionSpec[];
+}
+
 @inject("stores")
 @observer
 export class DocumentTabPanel extends BaseComponent<IProps, IState> {
 
+  subTabs: ISubTabSpec[] = [];
```

Comment:
Should this be in component state that is derived from the props?  I'm not sure if I'm following the pattern.

Reply:
The only reason to store it in state would be if we expect it to change and want such changes to trigger a re-render. Since it's static I think it's fine as an instance member that's initialized in the constructor which guarantees that it's complete before the first render.

Reply:
Ah, thanks for clarifying!